### ISSUE TITLE
Validation for nested properties not working

### ISSUE THREAD

**Issue by thnima** [Automattic/mongoose#8113]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I have the following schema:

```
const userSchema = new mongoose.Schema({
  billing: {
    recipient: {
      email: {
        type: String,
        lowercase: true,
        trim: true
      }
    }
  }
})

userSchema.path('billing.recipient.email').validate(function (email) {
  // do some validation stuff
}, 'ERROR')
```

When a user gets validated, the validator of `billing.recipient.email` doesn't get invoked.

It seems like this change in lib/document.js causes the issue: https://github.com/Automattic/mongoose/commit/6eeb48e1105c1bb0c701854edc8b91c84fa5a6e5#diff-fc7fc7567205f95715daa4204b8ea0b0.

Before this change, the whole path of the property - which is generated after the flatten function - looked like this: `billing.recipient.email`. Now it returns `billing.recipient.billing.recipient.email`

**What is the expected behavior?**
Validator should get invoked

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Mongoose: 5.6.9, 5.6.10
Node: v12.1.0

**Comment by vkarpov15** [comment]
The below script prints 'Called' 3 times for me, it looks like the validator runs for `validate()`, `validateSync()`, and [update validators](https://mongoosejs.com/docs/validation.html#update-validators). Can you please modify the below script to demonstrate your issue?

```javascript
const userSchema = new mongoose.Schema({
  billing: {
    recipient: {
      email: {
        type: String,
        lowercase: true,
        trim: true
      }
    }
  }
})

userSchema.path('billing.recipient.email').validate(function (email) {
  // do some validation stuff
}, 'ERROR')
```

**Comment by thnima** [comment]
Looks like this issue only happens, when I'm assigning a new object to a nested property. With the fix from Issue #8117 everything looks fine now.

Here is the script to reproduce:
```
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);

const GITHUB_ISSUE = `gh8113`;
const connectionString = `mongodb://@localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  await mongoose.connect(connectionString, { useNewUrlParser: true });
  await mongoose.connection.dropDatabase();

  const userSchema = new mongoose.Schema({
    billing: {
      recipient: {
        email: {
          type: String,
          lowercase: true,
          trim: true
        }
      }
    }
  })

  userSchema.path('billing.recipient.email').validate(function (email) {
    console.log('Called');
    return true;
  }, 'ERROR');

  const User = mongoose.model('User', userSchema);

  let doc = new User({ billing: { recipient: { email: 'test' } } });
  await doc.save()

  const user = await User.findById(doc._id)

  user.billing.recipient = {
    email: 'test2'
  }

  console.log('-----');
  await user.validate().catch(() => {});
  console.log('-----');
  user.validateSync();
  console.log('------');
  await User.findOneAndUpdate({}, { 'billing.recipient.email': 'foo' }, { runValidators: true }).catch(() => {});
}
```

**Comment by vkarpov15** [comment]
@thnima so it's fixed in 5.6.13?

**Comment by thnima** [comment]
Is fixed, yes
